src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeUnitRunner.java

```diff
@@ -96,56 +105,21 @@ public void run(RunNotifier notifier) {
     protected void runChild(FrameworkMethod method, RunNotifier notifier) {
 
         final Description description = describeChild(method);
+        JsonTestCase annotation = method.getMethod().getAnnotation(JsonTestCase.class);
 
         if (isIgnored(method)) {
-            notifier.fireTestIgnored(description);
-        } else {
-            JsonTestCase annotation = method.getMethod().getAnnotation(JsonTestCase.class);
-
-            if (annotation != null) {
-                currentTestCase = annotation.value();
-            } else {
-                currentTestCase = method.getName();
-            }
-
-            notifier.fireTestStarted(description);
-
-            LOGGER.debug("### Running currentTestCase : " + currentTestCase);
-
-            ScenarioSpec child = null;
-            try {
-                child = smartUtils.jsonFileToJava(currentTestCase, ScenarioSpec.class);
-
-                LOGGER.debug("### Found currentTestCase : -" + child);
-
-                final ZeroCodeMultiStepsScenarioRunner multiStepsRunner = getInjectedMultiStepsRunner();
-
-                /*
-                 * Override the properties file containing hosts and ports with HostProperties
-                 * only if the annotation is present on the runner.
-                 */
-                if (hostProperties != null) {
-                    ((ZeroCodeMultiStepsScenarioRunnerImpl) multiStepsRunner).overrideHost(host);
-                    ((ZeroCodeMultiStepsScenarioRunnerImpl) multiStepsRunner).overridePort(port);
-                    ((ZeroCodeMultiStepsScenarioRunnerImpl) multiStepsRunner).overrideApplicationContext(context);
-                }
-                passed = multiStepsRunner.runScenario(child, notifier, description);
 
-            } catch (Exception ioEx) {
-                ioEx.printStackTrace();
-                notifier.fireTestFailure(new Failure(description, ioEx));
-            }
-
-            testRunCompleted = true;
+            notifier.fireTestIgnored(description);
 
-            if (passed) {
-                LOGGER.info(String.format("\n**FINISHED executing all Steps for [%s] **.\nSteps were:%s",
-                        child.getScenarioName(),
-                        child.getSteps().stream().map(step -> step.getName()).collect(Collectors.toList())));
-            }
+        } else if (annotation != null) {
 
-            notifier.fireTestFinished(description);
+            runLeafJsonTest(notifier, description, annotation);
```

Comment:
👍

Reply:
No tests were found to make sure pure `JUnit tests` was executed. Please add a happy scenario

Reply:
yes, added one.
PR merged fyi. Please take care of the release to maven central.